src/components/App.jsx

```diff
@@ -43,6 +54,31 @@ export default function App() {
     setCurrentWalletInfo(null)
   }
 
+  // update maker/taker indicator based on websocket data
+  const onWebsocketMessage = useCallback((message) => {
+    const data = JSON.parse(message?.data)
+
+    // update the maker/taker indicator according to `coinjoin_state` property
+    if (data && typeof data.coinjoin_state === 'number') {
+      setCoinjoinInProcess(data.coinjoin_state === CJ_STATE_TAKER_RUNNING)
+      setMakerRunning(data.coinjoin_state === CJ_STATE_MAKER_RUNNING)
+    }
+  }, [])
+
+  useEffect(() => {
+    if (!websocket) return
+
+    websocket.addEventListener('message', onWebsocketMessage)
+
+    return () => websocket && websocket.removeEventListener('message', onWebsocketMessage)
+  }, [websocket, onWebsocketMessage])
+
+  // update the connection indicator based on the websocket connection state
+  useEffect(() => {
+    const websocketError = websocketState !== WebSocket.CONNECTING && websocketState !== WebSocket.OPEN
+    setWebsocketConnected(!websocketError)
```

Comment:
Technically we could also just do `setWebsocketConnected(websocketState === WebSocket.OPEN)` right? Since a connecting state isn't really connected. Just a nit though.